commit 1087f4f4af302e6e2fa40dd741f306444d90bece
Author: Francois Romieu <romieu@fr.zoreil.com>
Date:   Wed Dec 26 22:46:05 2007 +0100

    r8169: multicast register update (sync with Realtek's 8.004.00 8168 driver)
    
    The layout of the 8168 serie is different from that of the 8110 one.
    
    Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
    Cc: Edward Hsu <edward_hsu@realtek.com.tw>

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 6572425..9086e81 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -3098,8 +3098,10 @@ static void rtl_set_rx_mode(struct net_device *dev)
 	    (tp->mac_version == RTL_GIGA_MAC_VER_15) ||
 	    (tp->mac_version == RTL_GIGA_MAC_VER_16) ||
 	    (tp->mac_version == RTL_GIGA_MAC_VER_17)) {
-		mc_filter[0] = 0xffffffff;
-		mc_filter[1] = 0xffffffff;
+		u32 data = mc_filter[0];
+
+		mc_filter[0] = swab32(mc_filter[1]);
+		mc_filter[1] = swab32(data);
 	}
 
 	RTL_W32(MAR0 + 0, mc_filter[0]);
